networking_aci/plugins/ml2/drivers/mech_aci/allocations_manager.py

```diff
@@ -168,7 +187,7 @@ def _allocate_vxlan_segment(self, network, host_id, level, host_config):
             LOG.debug("Allocate %(type)s segment from pool failed with segment %(segment)s",
                       {"type": alloc.segment_type, "segment": alloc.segment_id, "level": alloc.level})
 
-            session.commit()
+            raise db_exc.RetryRequest(NoAllocationFoundInMaximumAllowedAttempts())
```

Comment:
maybe reuse neutron_lib.exceptions.ResourceExhausted for this case?

Reply:
We don't really expect to run into this exception, but `ResourceExhausted` doesn't sound right. The place where we look for exhausted resources is [here](https://github.com/sapcc/networking-aci/blob/986fa95e9212fdba5fce199307e61f4b575e03d4/networking_aci/plugins/ml2/drivers/mech_aci/allocations_manager.py#L136-L139)

But we could discuss, if `NoAllocationFoundInMaximumAllowedAttempts` should inherit from `neutron_lib.exceptions.ServiceUnavailable` as `neutron_lib.exceptions.NoNetworkFoundInMaximumAllowedAttempts` does.